packages/flutter/lib/src/widgets/dismissible.dart

```diff
@@ -691,3 +797,10 @@ class _DismissibleState extends State<Dismissible> with TickerProviderStateMixin
     );
   }
 }
+
+extension on DismissDirection {
+  bool get directionIsXAxis => switch (this) {
+    DismissDirection.horizontal || DismissDirection.endToStart || DismissDirection.startToEnd => true,
+    _ => false
+  };
+}
```

Comment:
A couple of relevant [style guidelines](https://github.com/flutter/flutter/blob/master/docs/contributing/Style-guide-for-Flutter-repo.md#summary):

#### avoid `extension`

Extension methods & getters are a bit more tricky to find, both in the source code and on [api.flutter.dev](https://api.flutter.dev/).

#### avoid `_`

This applies to inline callbacks (parameter names & types should be specified even when they aren't used) and also to [enum switch expressions](https://github.com/flutter/flutter/blob/master/docs/contributing/Style-guide-for-Flutter-repo.md#avoid-using-if-chains-or--or--with-enum-values).

The TL;DR is: we should put every enum value in the switch expression: that way, if a new value is added in the future, we can make sure it's handled correctly everywhere the enum is used.

Reply:
Luckily, it's an easy fix—just add it to the enum declaration:

```dart
enum DismissDirection {
  vertical,
  horizontal,
  endToStart,
  startToEnd,
  up,
  down,
  none;

  bool get directionIsXAxis => switch (this) {
    horizontal || endToStart || startToEnd => true,
    vertical   || up || down || none       => false,
  };
}
```